hibernate-core/src/test/java/org/hibernate/test/pagination/PaginationTest.java

```diff
@@ -118,6 +124,42 @@ public void testOffset() {
 		cleanupTestData();
 	}
 
+	/**
+	 * HHH-951: test that setting maxResults does not cause "ORA-00918: column ambiguously defined" on certain queries
+	 * 
+	 * @author Piotr Findeisen <piotr.findeisen@gmail.com>
+	 */
+	@Test
+	@RequiresDialectFeature(
+			value = DialectChecks.SupportLimitCheck.class,
+			comment = "Dialect does not support limit"
+	)
+	public void testLimitWithExpreesionAndFetchJoin() {
+		session = openSession();
+		session.beginTransaction();
+
+		String hql = "SELECT b, 1 FROM DataMetaPoint b inner join fetch b.dataPoint dp";
+		session.createQuery(hql)
+				.setMaxResults(3)
+				// This should not fail
+				.list();
+
+		HQLQueryPlan queryPlan = new HQLQueryPlan(hql, false, Collections.EMPTY_MAP, sessionFactory());
+		String sqlQuery = queryPlan.getTranslators()[0]
+				.collectSqlStrings().get(0);
+
+		session.getTransaction().commit();
+		session.close();
+
+		Matcher matcher = Pattern.compile(
+				"(?is)\\b(?<column>\\w+\\.\\w+)\\s+as\\s+(?<alias>\\w+)\\b.*\\k<column>\\s+as\\s+\\k<alias>")
+				.matcher(sqlQuery);
+		if (matcher.find()) {
+			fail(format("Column %s mapped to alias %s twice in generated SQL: %s", matcher.group("column"),
+					matcher.group("alias"), sqlQuery));
+		}
```

Comment:
If the column is mentioned twice in generated SQL wrapped in subquery, Oracle would complain with "ORA-00918: column ambiguously defined", so some kind of a check is necessary to prevent regression.

Whether this should be Pattern or something else -- I don't know. I wanted this to be as flexible as possible (so that this test doesn't need to be updated without real necessity), yet effective.

If you can think of a better way of checking this, I'll update the code.